### ISSUE TITLE
formula for KLD loss function in pretrain_vae.py

### ISSUE THREAD

**Issue by KunFang93** [CharlesNord/DEC-pytorch#1]
Hi,
I was wondering where this formula comes from: 
KLD = -0.5 * torch.sum(1 + logvar - mu.pow(2) - logvar.exp())
KLD /= 784 * 128
thanks a lot!

**Comment by CharlesNord** [comment]
> Hi,
> I was wondering where this formula comes from:
> KLD = -0.5 * torch.sum(1 + logvar - mu.pow(2) - logvar.exp())
> KLD /= 784 * 128
> thanks a lot!

这是变分自编码器的内容，用于约束隐含层变量的分布接近正态分布

**Comment by KunFang93** [comment]
哦哦好的，不过我看文章里，
<img width="811" alt="Screen Shot 2019-08-30 at 2 56 36 PM" src="https://user-images.githubusercontent.com/15643651/64048117-65e9bb80-cb36-11e9-8564-00aacc8ad4e9.png">
这里好像并没有处以batch size， 能问一下code里的KLD /= 784*128 是怎么来的么？
谢谢啦！

**Comment by CharlesNord** [comment]
> 哦哦好的，不过我看文章里，
> <img alt="Screen Shot 2019-08-30 at 2 56 36 PM" width="811" src="https://user-images.githubusercontent.com/15643651/64048117-65e9bb80-cb36-11e9-8564-00aacc8ad4e9.png">
> 这里好像并没有处以batch size， 能问一下code里的KLD /= 784*128 是怎么来的么？
> 谢谢啦！

这个不要在意，加不加都行。当时想的是计算 reconstruction loss 的时候，算的是平均loss，所以把 KLD 也除以所有像素。实际上 reconstruction loss 和 KLD loss 之间的权重是可以调的，不一定非要 1:1

**Comment by KunFang93** [comment]
哦哦哦，这样子，谢谢啦！
